### PR TITLE
refactor(context): inject utils methods through context and improve typing on some files

### PR DESCRIPTION
**Pull request by DrRaider**
### General

- [ ] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [ ] Test manually the implemented changes
- [ ] Validate the code quality (indentation, syntax, style, simplicity, readability)

### Security

- [ ] Consider the security impact of the changes made

### CODE REVIEW

File: src/context/dependencies-plan.js [hunk 1]
```diff
@@ -24,5 +24,6 @@ module.exports = plan =>
         .addModule('openIdClient', () => require('openid-client'))
         .addModule('os', () => require('os'))
         .addModule('superagent', () => require('superagent'))
-        .addUsingFunction('diffString', () => require('json-diff').diffString),
+        .addUsingFunction('diffString', () => require('json-diff').diffString)
+        .addModule('lodash', () => require('lodash')),
```
Comment (SteveBunlon) [inline src/context/dependencies-plan.js:28]: If this comes to the context, every usage of lodash should use the context as-well. 

A [PR](https://github.com/ForestAdmin/toolbelt/pull/546/files) is ongoing for that so let's skip

File: test/services/dumpers/abstract-dumper.unit.test.ts [hunk 2]
```diff
@@ -27,6 +27,7 @@ const buildDumper = (ADumper, librairies = {}) => {
       compile: jest.fn().mockImplementation(() => () => {}),
     },
     buildDatabaseUrl: jest.fn(),
+    toValidPackageName: jest.fn().mockImplementation(content => content),
```
Comment (SteveBunlon) [inline test/services/dumpers/abstract-dumper.unit.test.ts:30]: This should be removed

File: src/services/dumpers/forest-express.js [hunk 3]
```diff
@@ -2,7 +2,6 @@ const _ = require('lodash');
 const { URL } = require('url');
 const { plural, singular } = require('pluralize');
 const stringUtils = require('../../utils/strings');
```
Comment (SteveBunlon) [inline src/services/dumpers/forest-express.js:4]: This should be removed

File: test/utils/database-url.unit.test.js [hunk 4]
```diff
@@ -72,15 +35,72 @@ describe('utils > buildDatabaseUrl', () => {
...
+              mongodbSrv: true,
+            };
+            const databaseUrl = buildDatabaseUrl(config);
+
+            expect(databaseUrl).toBe(
+              'mongodb+srv://__dbUser__:__dbPassword__@__dbHostname__/__dbName__',
+            );
+          });
+
+          it('should ignore it otherwise', () => {
```
Comment (SteveBunlon) [inline test/utils/database-url.unit.test.js:66]: This is not correct. You are not testing ONLY the srv option here, you are also testing that the conenctionUrl is correctly built, whatever the dialect. 

This comes from an upper test you relocated. It should stay as it was before `should build the URL from the configuration`

File: tsconfig.json [hunk 5]
```diff
@@ -7,6 +7,7 @@
     "skipLibCheck": true,
     "module": "CommonJS",
     "moduleResolution": "Node",
+    "resolveJsonModule": true,
```
Comment (SteveBunlon) [inline tsconfig.json:10]: Used to import .JSON files 🆗

**Review by SteveBunlon — CHANGES_REQUESTED** [review]

File: src/context/dependencies-plan.js (hunk 1 above)
Comment (DrRaider) [reply src/context/dependencies-plan.js:28]: 👍

File: tsconfig.json (hunk 5 above)
Comment (DrRaider) [reply tsconfig.json:10]: 👍

**Comment by SteveBunlon** [comment]
Functionnal tests: 
- [ ] create project SQL

- [ ] create project NOSQL

- [ ] schema:update

- [ ] schema:apply

**Pull request merged**